File: src/main/java/com/docs/viewer/global/common/utils/FileUtil.java
package com.docs.viewer.global.common.utils;

import com.docs.viewer.global.common.setting.FileSetting;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Component;

import jakarta.annotation.PostConstruct;
import java.io.File;
import java.nio.file.Path;
import java.nio.file.Paths;
import java.util.UUID;

@Component
@RequiredArgsConstructor
public class FileUtil {

    private static final String PROJECT_DIRECTORY = System.getProperty("user.dir");


    public static Path getDirectory(String filepath, String targetFolder) {
        String directoryPath =  targetFolder + "/" + DateUtil.getCurrentDate();
        Path folderPath = Paths.get(filepath);
        return folderPath.resolve(directoryPath);
    }

    public static Path getUploadDirectory(Path uploadDirectory) {
        createDirectory(uploadDirectory.toString()); // 업로드할 디렉토리가 없으면 생성
        return uploadDirectory;
    }

    public static File getTargetFile(String targetFilePath) {
        return new File(PROJECT_DIRECTORY + "/" + targetFilePath);
    }

    public static String getFilePath(Path uploadDirectory) {
        return uploadDirectory.toString() + "/" + UUID.randomUUID();
    }

    public static String getFileExtension(String fileName) {
        int lastDotIndex = fileName.lastIndexOf(".");
        if (lastDotIndex > 0) {
            return fileName.substring(lastDotIndex + 1);
        }
        return "";
    }

    public static String removeFileExtension(String fileName) {
        int lastDotIndex = fileName.lastIndexOf(".");
        if (lastDotIndex > 0) {
            return fileName.substring(0, lastDotIndex);
        }
        return fileName;
    }

    private static void createDirectory(String uploadDirectory) {
        File directory = new File(uploadDirectory);
        if (!directory.exists()) {
            directory.mkdirs();
        }
    }
}

File: dotnet-docs-viewer/Common/Utils/FileUtil.cs
namespace Common.Utils
{
    public static class FileUtil
    {
        private static readonly string PROJECT_DIRECTORY = Directory.GetCurrentDirectory();

        /// <summary>업로드 디렉토리 경로 생성 (날짜 기준)</summary>
        public static string GetDirectory(string filepath, string targetFolder)
        {
            var directoryPath = Path.Combine(
                targetFolder,
                DateUtil.GetCurrentDate()
            );

            return Path.Combine(filepath, directoryPath);
        }

        /// <summary>업로드 디렉토리 생성 (없으면 생성)</summary>
        public static string GetUploadDirectory(string uploadDirectory)
        {
            CreateDirectory(uploadDirectory);
            return uploadDirectory;
        }

        /// <summary>실제 파일 객체 반환</summary>
        public static FileInfo GetTargetFile(string targetFilePath)
        {
            return new FileInfo(
                Path.Combine(PROJECT_DIRECTORY, targetFilePath)
            );
        }

        /// <summary>UUID 기반 파일 경로 반환</summary>
        public static string GetFilePath(string uploadDirectory)
        {
            return Path.Combine(
                uploadDirectory,
                Guid.NewGuid().ToString()
            );
        }

        /// <summary>파일 확장자 반환</summary>
        public static string GetFileExtension(string fileName)
        {
            return Path.GetExtension(fileName)
                .TrimStart('.');
        }

        /// <summary>확장자 제거</summary>
        public static string RemoveFileExtension(string fileName)
        {
            return Path.GetFileNameWithoutExtension(fileName);
        }

        /// <summary>디렉토리 생성</summary>
        private static void CreateDirectory(string uploadDirectory)
        {
            if (!Directory.Exists(uploadDirectory))
            {
                Directory.CreateDirectory(uploadDirectory);
            }
        }
    }
}